compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIELowerToAIE.cpp

```diff
@@ -0,0 +1,802 @@
+                          mapper.lookup(out.getDefiningOp()))
+                          .getSymName();
+    outSyms.push_back(SymbolRefAttr::get(rewriter.getContext(), symbolName));
+  }
+  rewriter.create<AIE::ObjectFifoLinkOp>(rewriter.getUnknownLoc(),
+                                         rewriter.getArrayAttr(inSyms),
+                                         rewriter.getArrayAttr(outSyms));
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// Convert amdaie.tile operation to aie.tile
+//===----------------------------------------------------------------------===//
+
+LogicalResult tileToAIE(IRRewriter &rewriter, AMDAIE::TileOp tileOp,
+                        IRMapping &mapper, Block *deviceBlock) {
+  LLVM_DEBUG(llvm::dbgs() << "Convert [AMDAIE::TileOp]\n");
+  OpBuilder::InsertionGuard guard(rewriter);
+  int64_t col = getConstantIntValue(tileOp.getCol()).value();
+  int64_t row = getConstantIntValue(tileOp.getRow()).value();
+  rewriter.setInsertionPointToStart(deviceBlock);
+  auto aieTileOp =
+      rewriter.create<xilinx::AIE::TileOp>(rewriter.getUnknownLoc(), col, row);
+  mapper.map(tileOp.getResult(), aieTileOp.getResult());
+  mapper.map(tileOp.getOperation(), aieTileOp.getOperation());
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// Convert amdaie.workgroup operation and insert into aie.device
+//===----------------------------------------------------------------------===//
+
+LogicalResult workgroupToAIE(IRRewriter &rewriter,
+                             AMDAIE::WorkgroupOp workgroupOp,
+                             xilinx::AIE::DeviceOp deviceOp,
+                             func::FuncOp ipuFuncOp, IRMapping &mapper,
+                             IRMapping &bindingsMapper) {
+  OpBuilder::InsertionGuard guard(rewriter);
+  Block *deviceBlock = &deviceOp.getRegion().front();
+  Block *deviceCoreBlock = rewriter.createBlock(&deviceOp.getRegion());
+  rewriter.setInsertionPoint(deviceBlock, deviceBlock->begin());
+
+  // Walk all operations in the AIE region and convert to AIE ops
+  int dmaId = 0;
+  WalkResult res = workgroupOp.walk<WalkOrder::PreOrder>([&](Operation *op) {
+    return TypeSwitch<Operation *, WalkResult>(op)
+        .Case<AMDAIE::CircularDmaCpyNdOp>([&](auto dmaOp) {
+          if (failed(circularDmaToAIE(rewriter, dmaOp, mapper, deviceBlock,
+                                      dmaId))) {
+            return WalkResult::interrupt();
+          }
+          return WalkResult::advance();
+        })
+        .Case<AMDAIE::ControlCodeOp>([&](auto controlCodeOp) {
+          if (failed(controlCodeToAie(rewriter, controlCodeOp, ipuFuncOp,
+                                      mapper, bindingsMapper))) {
+            controlCodeOp.emitError("could not convert to AIEDialect ops");
+            return WalkResult::interrupt();
+          }
+          return WalkResult::skip();
```

Comment:
Why do you use `WalkResult::skip()` not `WalkResult::advance()` here?

Reply:
To avoid traversing the body of the control code operation as it's handled separately.